core/src/main/java/com/orientechnologies/orient/core/db/document/ODatabaseDocumentEmbedded.java

```diff
@@ -133,6 +126,22 @@
   private OrientDBConfig config;
   private OStorage storage;
 
+  InterruptTimerTask commandInterruptTimer;
+
+  class InterruptTimerTask extends TimerTask {
+
+    Thread executionThread;
```

Comment:
Can we have the private/protected visibilities as needed

Reply:
Yep, definitely. The commands have their own management of timeout (see below, OSelectExecutionPlanner) but there can still be long running scripts. 
So yes, it makes sense

Reply:
Done